tests/api.c

```diff
@@ -5583,6 +5589,760 @@ static int test_wc_ChaCha20Poly1305_aead (void)
+        0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37,
+        0x38, 0x39, 0x61, 0x62, 0x63, 0x64, 0x65
+    };
+    byte    iv[]    = "1234567890abcdef";
+    int     ret;
+
+    printf(testingFmt, "wc_AesSetKey()");
+
+    ret = wc_AesInit(&aes, NULL, INVALID_DEVID);
+    if (ret == 0) {
+        ret = wc_AesSetKey(&aes, key16, (word32) sizeof(key16) / sizeof(byte),
+                                                        iv, AES_ENCRYPTION);
+    }
+    if (ret == 0) {
+        ret = wc_AesSetKey (&aes, key24, (word32) sizeof(key24) / sizeof(byte),
+                                                           iv, AES_ENCRYPTION);
+    }
+    if (ret == 0) {
+        ret = wc_AesSetKey (&aes, key32, (word32) sizeof(key32) / sizeof(byte),
+                                                           iv, AES_ENCRYPTION);
+    }
+    /* Pass in bad args. */
+    if (ret == 0) {
+        ret = wc_AesSetKey (NULL, key16, (word32) sizeof(key16) / sizeof(byte),
+                                                           iv, AES_ENCRYPTION);
+            if (ret == BAD_FUNC_ARG) {
+                ret = wc_AesSetKey(&aes, badKey16,
+                                    (word32) sizeof(badKey16) / sizeof(byte),
+                                                         iv, AES_ENCRYPTION);
+        }
+        if (ret == BAD_FUNC_ARG) {
+            ret = 0;
+        } else {
+            ret = SSL_FATAL_ERROR;
+        }
+    }
+
+    printf(resultFmt, ret == 0 ? passed : failed);
+
+#endif
+    return 0;
+} /* END test_wc_AesSetKey */
+
+
+
+/*
+ * test function for wc_AesCbcEncrypt(), wc_AesCbcDecrypt(),
+ * and wc_AesCbcDecryptWithKey()
+ */
+static int test_wc_AesCbcEncryptDecrypt (void)
+{
+#if !defined(NO_AES) && defined(HAVE_AES_CBC) && defined(HAVE_AES_DECRYPT)
+    Aes     aes;
+    byte    key32[] =
+    {
+        0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37,
+        0x38, 0x39, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66,
+        0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37,
+        0x38, 0x39, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66
+    };
+    byte    vector[] = /* Now is the time for all w/o trailing 0 */
+    {
+        0x4e,0x6f,0x77,0x20,0x69,0x73,0x20,0x74,
+        0x68,0x65,0x20,0x74,0x69,0x6d,0x65,0x20,
+        0x66,0x6f,0x72,0x20,0x61,0x6c,0x6c,0x20
+    };
+    byte    iv[]    = "1234567890abcdef";
+    byte    enc[AES_BLOCK_SIZE];
+    byte    dec[sizeof(vector)];
+    int     cbcE    =   SSL_FATAL_ERROR;
+    int     cbcD    =   SSL_FATAL_ERROR;
+    int     cbcDWK  =   SSL_FATAL_ERROR;
+    byte    dec2[sizeof(vector)];
+    int     ret;
+
+    /* Init stack variables. */
+    XMEMSET(enc, 0, AES_BLOCK_SIZE);
+    XMEMSET(dec, 0, sizeof(vector));
+    XMEMSET(dec2, 0, sizeof(vector));
+
+    ret = wc_AesInit(&aes, NULL, INVALID_DEVID);
+    if (ret == 0) {
+        ret = wc_AesSetKey(&aes, key32, AES_BLOCK_SIZE * 2, iv, AES_ENCRYPTION);
+    }
+    if (ret == 0) {
+        ret = wc_AesCbcEncrypt(&aes, enc, vector, sizeof(vector));
```

Comment:
Should these AesCbcEncrypt/Decrypt() calls be after the printf() for wc_AesCbcEncrypt?